File: FTC/2023/22364/java/src/org/firstinspires/ftc/teamcode/POWERPLAYStarterBotTeleop.java
package org.firstinspires.ftc.teamcode;

import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
import com.qualcomm.robotcore.hardware.CRServo;
import com.qualcomm.robotcore.hardware.DcMotor;
import com.qualcomm.robotcore.hardware.DcMotorSimple;

@TeleOp(name = "POWERPLAYStarterBotTeleop (Blocks to Java)")
public class POWERPLAYStarterBotTeleop extends LinearOpMode {

  private DcMotor leftarmmotor;
  private DcMotor right_drive;
  private DcMotor left_drive;
  private DcMotor rightarmmotor;
  private CRServo IntakeAsCRServo;

  /**
   * This function is executed when this Op Mode is selected from the Driver Station.
   */
  @Override
  public void runOpMode() {
    leftarmmotor = hardwareMap.get(DcMotor.class, "left arm motor");
    right_drive = hardwareMap.get(DcMotor.class, "right_drive");
    left_drive = hardwareMap.get(DcMotor.class, "left_drive");
    rightarmmotor = hardwareMap.get(DcMotor.class, "right arm motor");
    IntakeAsCRServo = hardwareMap.get(CRServo.class, "IntakeAsCRServo");

    // You will need to reverse one of the drive motors.
    // You will have to determine which motor to reverse for your robot.
    // In this example, the right motor was reversed, so that positive
    // applied power makes it move the robot in the forward direction.
    leftarmmotor.setDirection(DcMotorSimple.Direction.REVERSE);
    right_drive.setDirection(DcMotorSimple.Direction.REVERSE);
    waitForStart();
    if (opModeIsActive()) {
      // Put your run blocks here.
      while (opModeIsActive()) {
        // Put your loop blocks here.
        // Use left stick to drive and right stick to turn
        // The Y axis of a joystick ranges from -1 in its topmost
        // position to +1 in the bottom position. We negate (i.e.
        // reverse the positive/negative sign of) these values, so that
        // pushing the joystick up will create maximum forward power.
        left_drive.setPower(-gamepad1.left_stick_y - gamepad1.right_stick_x);
        right_drive.setPower(-gamepad1.left_stick_y + gamepad1.right_stick_x);
        // If DPadUp is pressed, lift raises.
        if (gamepad1.dpad_up) {
          leftarmmotor.setPower(-1);
          rightarmmotor.setPower(-1);
        } else {
          // If DPadDown is pressed, lift lowers.
          if (gamepad1.dpad_down) {
            leftarmmotor.setPower(0.8);
            rightarmmotor.setPower(0.8);
          } else {
            // If DPad is not pressed, lift remains still.
            leftarmmotor.setPower(0);
            rightarmmotor.setPower(0);
          }
        }
        // If RightBumper is pressed, Intake receives power
        if (gamepad1.right_bumper) {
          IntakeAsCRServo.setPower(-1);
        } else {
          IntakeAsCRServo.setPower(0.3);
        }
        // Code below details data collection during OpMode
        telemetry.addData("Left Pow", right_drive.getPower());
        telemetry.addData("Right Pow", left_drive.getPower());
        telemetry.update();
      }
    }
  }
}

File: FTC/2023/22364/blocks/POWERPLAY-Starter-Bot-Teleop.js
// IDENTIFIERS_USED=gamepad1,intakeservoAsCRServo,left_driveAsDcMotor,leftarmmotorAsDcMotor,right_driveAsDcMotor,rightarmmotorAsDcMotor

var turnSpeed, driveSpeed, armSpeed;

/**
 * This function is executed when this Op Mode is selected from the Driver Station.
 */
function runOpMode() {
  // You will have to determine which motor to reverse for your robot.
  // In this example, the right motor was reversed, so that positive
  // applied power makes it move the robot in the forward direction.
  leftarmmotorAsDcMotor.setDirection("REVERSE");
  left_driveAsDcMotor.setDirection("REVERSE");
  turnSpeed = 0.6;
  driveSpeed = 0.8;
  armSpeed = -0.8;
  linearOpMode.waitForStart();
  if (linearOpMode.opModeIsActive()) {
    while (linearOpMode.opModeIsActive()) {
      // The Y axis of a joystick ranges from -1 in its topmost
      // position to +1 in the bottom position. We negate (i.e.
      // reverse the positive/negative sign of) these values, so that
      // pushing the joystick up will create maximum forward power.
      left_driveAsDcMotor.setDualPower(-(gamepad1.getLeftStickY() * driveSpeed) + gamepad1.getRightStickX() * turnSpeed, right_driveAsDcMotor, -(gamepad1.getLeftStickY() * driveSpeed) - gamepad1.getRightStickX() * turnSpeed);
      if (gamepad1.getDpadDown()) {
        leftarmmotorAsDcMotor.setDualPower(0.1, rightarmmotorAsDcMotor, 0.1);
      } else {
        if (gamepad1.getDpadUp()) {
          leftarmmotorAsDcMotor.setPower(armSpeed);
          rightarmmotorAsDcMotor.setPower(armSpeed);
        } else {
          leftarmmotorAsDcMotor.setPower(-0.15);
          rightarmmotorAsDcMotor.setPower(-0.15);
        }
      }
      if (gamepad1.getRightBumper()) {
        intakeservoAsCRServo.setPower(-1);
      } else if (gamepad1.getLeftBumper()) {
        intakeservoAsCRServo.setPower(0.8);
      } else {
        intakeservoAsCRServo.setPower(0.6);
      }
      telemetry.addNumericData('Left Pow', right_driveAsDcMotor.getPower());
      telemetry.addNumericData('Right Pow', left_driveAsDcMotor.getPower());
      telemetry.update();
      telemetry.addNumericData('driveSpeed', driveSpeed);
      telemetry.addNumericData('turnSpeed', turnSpeed);
    }
  }
}
